adaptive_scheduler/kernel/slicedipscheduler_v2.py

```diff
@@ -163,11 +169,18 @@ def get_slices(self, intervals, resource, duration):
                         start += slice_length
                         internal_start = start
 
+            # Get the airmass coefficients for the internal starts if the request should be optimized by airmass
+            if request and request.optimization_type == 'AIRMASS':
+                airmasses_at_times = request.get_airmasses_within_kernel_windows(resource)
+                # use numpy to interpolate airmass values for the internal_starts times
+                interpolated_airmasses = np.interp(internal_starts, airmasses_at_times['times'], airmasses_at_times['airmasses'])
```

Comment:
Here is where we pull down the cached airmass values at unix times and interpolate them to get values for the internal start times of each possible scheduling window.